customers/schema/types.py

```diff
@@ -65,39 +70,68 @@ class Meta:
         connection_class = CountConnection
 
 
-PROFILE_NODE_FIELDS = (
-    "id",
-    "invoicing_type",
-    "comment",
-    "organization",
-    "boats",
-    "berth_applications",
-    "berth_leases",
-    "orders",
-)
-
-
 class BerthApplicationFilter(django_filters.FilterSet):
     order_by = django_filters.OrderingFilter(
         fields=(("created_at", "createdAt"),),
         label="Supports `createdAt` and `-createdAt`.",
     )
 
 
-class BaseProfileFieldsMixin:
+class ProfileFilterSet(django_filters.FilterSet):
+    class Meta:
+        model = CustomerProfile
+        fields = ("comment",)
+
+    comment = django_filters.CharFilter(lookup_expr="icontains")
+
+
+def _return_profile_if_user_has_permissions(profile, user):
+    if profile.user == user or user_has_view_permission(
+        user, CustomerProfile, BerthApplication, BerthLease
+    ):
+        return profile
+    else:
+        raise VenepaikkaGraphQLError(
+            _("You do not have permission to perform this action.")
+        )
```

Comment:
this is a bit tricky, since here node is already profile, which is the only model directly connected to the user instance. other models have `customer` field, which is the profile, so there is that extra level. but maybe something like this could work:
```python
def _return_node_if_user_has_permissions(node, user, *models):
    node_user = node.user if hasattr(node, user) else node.customer.user
    if node_user == user or user_has_view_permission(user, *models):
        return node
    else:
        raise VenepaikkaGraphQLError(
            _("You do not have permission to perform this action.")
        )
```
🤔 🤔 🤔

Reply:
Just adding the check if `node == None` and that should do the trick, since it's required also for leases

```python
def _return_node_if_user_has_permissions(node, user, *models):
    if not node:
        return None

    node_user = node.user if hasattr(node, user) else node.customer.user
    if node_user == user or user_has_view_permission(user, *models):
        return node
    else:
        raise VenepaikkaGraphQLError(
            _("You do not have permission to perform this action.")
        )
```

Reply:
yeah, good catch. could you give some examples of where you would need this and what would be `node` and `models` in those cases?

Reply:
The case where I was trying to work this out was this:

https://github.com/City-of-Helsinki/berth-reservations/blob/develop/leases/schema.py#L61-L74
```python
    def get_node(cls, info, id):
        node = super().get_node(info, id)
        if not node:
            return None

        user = info.context.user
        if (node.customer and node.customer.user == user) or user_has_view_permission(
            user, BerthLease, BerthApplication, CustomerProfile
        ):
            return node
        else:
            raise VenepaikkaGraphQLError(
                _("You do not have permission to perform this action.")
            )
```

Which then would be:
```python
    def get_node(cls, info, id):
        node = super().get_node(info, id)

        user = info.context.user
        return _return_node_if_user_has_permissions(
            node, user, BerthLease, BerthApplication, CustomerProfile
        )
```

Reply:
alright, implemented an MVP for the util 😎

Reply:
⭐ ⭐ ⭐ ⭐ ⭐ 
Would definitely recommend